AssertionError when generating annotations

coverage==3.7.1
pytest==2.8.0
pytest-cov==2.1.0

I run `py.test --cov-report annotate --cov=<module> <py file>` and I get

```
  File "/Users/c17r/.virtualenvs/<project>/lib/python2.7/site-packages/pytest_cov/plugin.py", line 172, in pytest_terminal_summary
    assert total is not None, 'Test coverage should never be `None`'
_pytest.assertion.reinterpret.AssertionError: Test coverage should never be `None`
```

The annotated source is created with `<py file>,cover` so it's usable but the stack is big enough (and my window small enough) that I have to keep scrolling up to see test results

if I run `--cov-report term-missing` instead then there is no stack trace but, obviously, no annotations

--- a/src/pytest_cov/engine.py
+++ b/src/pytest_cov/engine.py
@@ -91,7 +91,8 @@ class CovController(object):
 
         # Produce annotated source code report if wanted.
         if 'annotate' in self.cov_report:
-            total = self.cov.annotate(ignore_errors=True)
+            self.cov.annotate(ignore_errors=True)
+            total = self.cov.report(ignore_errors=True, file=StringIO())
             stream.write('Coverage annotated source written next to source\n')
 
         # Produce html report if wanted.

--- a/src/pytest_cov/engine.py
+++ b/src/pytest_cov/engine.py
@@ -92,6 +92,8 @@ class CovController(object):
         # Produce annotated source code report if wanted.
         if 'annotate' in self.cov_report:
             self.cov.annotate(ignore_errors=True)
+            # We need to call Coverage.report here, just to get the total
+            # Coverage.annotate don't return any total and we need it for --cov-fail-under.
             total = self.cov.report(ignore_errors=True, file=StringIO())
             stream.write('Coverage annotated source written next to source\n')
 

--- a/tests/test_pytest_cov.py
+++ b/tests/test_pytest_cov.py
@@ -154,6 +154,22 @@ def test_central(testdir):
     assert result.ret == 0
 
 
+def test_annotate(testdir):
+    script = testdir.makepyfile(SCRIPT)
+
+    result = testdir.runpytest('-v',
+                               '--cov=%s' % script.dirpath(),
+                               '--cov-report=annotate',
+                               script)
+
+    result.stdout.fnmatch_lines([
+        '*- coverage: platform *, python * -*',
+        'Coverage annotated source written next to source',
+        '*10 passed*',
+    ])
+    assert result.ret == 0
+
+
 def test_cov_min_100(testdir):
     script = testdir.makepyfile(SCRIPT)
 

